bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/connection/Clip2Bridge.java

```diff
@@ -0,0 +1,980 @@
+    }
+
+    /**
+     * Send a ping to the Hue bridge to check that the session is still alive.
+     */
+    private void checkAlive() {
+        if (onlineState == State.CLOSED) {
+            return;
+        }
+        logger.debug("checkAlive()");
+        Session session = http2Session;
+        if (Objects.nonNull(session)) {
+            session.ping(new PingFrame(false), Callback.NOOP);
+        }
+        if (Instant.now().isAfter(sessionExpireTime)) {
+            fatalError(this, Http2Error.TIMEOUT);
+        }
+    }
+
+    /**
+     * Connection is ok, so reschedule the session check alive expire time. Called in response to incoming ping frames
+     * from the bridge.
+     */
+    protected void checkAliveOk() {
+        logger.debug("checkAliveOk()");
+        sessionExpireTime = Instant.now().plusSeconds(CHECK_ALIVE_SECONDS * 2);
+    }
+
+    /**
+     * Close the connection.
+     */
+    @Override
+    public void close() {
+        synchronized (this) {
+            logger.debug("close()");
+            boolean notifyHandler = (onlineState == State.ACTIVE) && !restarting;
+            onlineState = State.CLOSED;
+            closeCheckAliveTask();
+            closeSession();
+            try {
+                http2Client.stop();
+            } catch (Exception e) {
+                // ignore
+            }
+            if (notifyHandler) {
+                bridgeHandler.onConnectionOffline();
+            }
+        }
+    }
+
+    /**
+     * Close the check alive task if necessary.
+     */
+    private void closeCheckAliveTask() {
+        logger.debug("closeCheckAliveTask()");
+        ScheduledFuture<?> task = checkAliveTask;
+        if (Objects.nonNull(task)) {
+            task.cancel(true);
+        }
+        checkAliveTask = null;
+    }
+
+    /**
+     * Close the HTTP 2 session if necessary.
+     */
+    private void closeSession() {
+        logger.debug("closeSession()");
+        Session session = http2Session;
+        if (Objects.nonNull(session)) {
+            session.close(0, null, Callback.NOOP);
+        }
+        http2Session = null;
+    }
+
+    /**
+     * Method that is called back in case of fatal stream or session events. Note: under normal operation, the Hue
+     * Bridge sends a 'soft' GO_AWAY command every nine or ten hours, so we handle such soft errors by attempting to
+     * silently close and re-open the connection without notifying the handler of an actual 'hard' error.
+     *
+     * @param cause the entity that caused this method to be called.
+     * @param error the type of error.
+     */
+    private void fatalError(Object cause, Http2Error error) {
+        if (restarting || (onlineState == State.CLOSED)) {
+            return;
+        }
+        String causeId = cause.getClass().getSimpleName();
+        if (cause instanceof ContentStreamListenerAdapter) {
+            logger.debug("fatalError() {} {} ignoring", causeId, error);
```

Comment:
Maybe add a comment why we can ignore it here? `// content fetch errors are handled by handling exceptions happening while fetching the result already`

Reply:
Ok but I will use a better text.